src/main/java/de/tum/cit/aet/artemis/programming/service/ProgrammingExerciseService.java

```diff
@@ -427,6 +429,26 @@ private void validatePackageName(ProgrammingExercise programmingExercise, Progra
         }
     }
 
+    private void validateCustomCheckoutPaths(ProgrammingExercise programmingExercise) {
+        var buildConfig = programmingExercise.getBuildConfig();
+
+        boolean assignmentCheckoutPathIsValid = isValidCheckoutPath(buildConfig.getAssignmentCheckoutPath());
+        boolean solutionCheckoutPathIsValid = isValidCheckoutPath(buildConfig.getSolutionCheckoutPath());
+        boolean testCheckoutPathIsValid = isValidCheckoutPath(buildConfig.getTestCheckoutPath());
+
+        if (!assignmentCheckoutPathIsValid || !solutionCheckoutPathIsValid || !testCheckoutPathIsValid) {
+            throw new BadRequestAlertException("The custom checkout paths are invalid", "Exercise", "checkoutDirectoriesInvalid");
+        }
+    }
+
+    private boolean isValidCheckoutPath(String checkoutPath) {
+        if (checkoutPath == null) {
```

Comment:
NIT: I think it would be helpful to add a short comment here. Initially, it wasn't clear to me why a null checkoutPath would be considered valid.

Reply:
Depending on your preferences we could also use Optional or @Nullable to make it clearer that they do not have to be set